src/issuance.rs

```diff
@@ -480,6 +484,34 @@ impl IssueBundle<Unauthorized> {
         Ok(())
     }
 
+    /// Compute the correct rho value for each note in the bundle according to
+    /// [ZIP-227: Issuance of Zcash Shielded Assets][zip227].
+    ///
+    /// [zip227]: https://zips.z.cash/zip-0227
+    pub fn update_rho(self, first_nullifier: Nullifier) -> IssueBundle<PartiallyPrepared> {
```

Comment:
We don't need to copy the Nullifier for this.
Consider: 
```rust
pub fn update_rho(self, first_nullifier: &Nullifier)
```
all the way down (`update_rho_for_issuance_note` and `rho_for_issuance_note`).

Reply:
Done